cache/class-vary-cache.php

```diff
@@ -86,6 +90,11 @@ public static function set_group_for_user( $group, $value ) {
 		// TODO: only send header if we added or changed things
 		// TODO: don't set the cookie if was already set on the request
 		// validate, process $group, etc.
+		if ( strpos( $group, self::GROUP_SEPARATOR ) !== false || strpos( $group, self::VALUE_SEPARATOR ) !== false ||
+			strpos( $value, self::GROUP_SEPARATOR ) !== false || strpos( $value, self::VALUE_SEPARATOR ) !== false ) {
```

Comment:
It would be good to split out the validation for group vs value so we can provide a more specific error message.